### ISSUE TITLE
Deserializing resource with no relationships included

### ISSUE THREAD

**Issue by jimmyrose** [codecutout/JsonApiSerializer#84]
I have an example json api resource that looks like this:

`{
  "data": [
    {
      "attributes": {
        "name": "Test",
        "description": "Unassigned"
      },
      "relationships": {
        "areas": {
          "links": {
            "self": "https://localhost:44317/api/types/1/relationships/areas",
            "related": "https://localhost:44317/api/types/1/areas"
          }
        }
      },
      "type": "types",
      "id": "1"
    }
  ]
}`

Note that there is no 'data' in the relationship - this is intentional because we don't want to include them in this request to the server.

I get an exception: Expected to find nested object within data[0].relationships.areas that matches \data\

If I'm reading this right, it sounds like the deserializer is expecting "data". The JSON API spec (https://jsonapi.org/format/#document-resource-object-relationships) says it must contain **at least one** of the following: links, data, meta. 

So I believe in this case data should not be required, because 'links' are present. In which case the relationships should just be empty. Could be totally wrong though ¯\_(ツ)_/¯

**Comment by alex-davies** [comment]
You are absolutely right, dataless relationships are totally fine

Because the `links` and `meta` belong to the relationship rather than the resource object you have to have the relationship object in your .net model. To do this you can either use the provided classes of `Relationship` or `Relationship<TData>`, alternatively you can also define your own relationship class, you just need to have either a `links` or `data` property present on it.

Example of using the `Relationship` object

```csharp
[Fact]
public void When_dataless_relationship_should_deserialize()
{
    var json = @"
{  
    ""data"": {
    ""attributes"":{
        ""name"":""Test"",
        ""description"":""Unassigned""
    },
    ""relationships"":{
        ""areas"":{
            ""links"":{
                ""self"":""https://localhost:44317/api/types/1/relationships/areas"",
                ""related"":""https://localhost:44317/api/types/1/areas""
            }
        }
    },
    ""type"":""types"",
    ""id"":""1""
    }
}";
    var obj = JsonConvert.DeserializeObject<Types>(json, new JsonApiSerializerSettings());

    Assert.Equal("Test", obj.Name);
    Assert.Equal("Unassigned", obj.Description);
    Assert.Equal("https://localhost:44317/api/types/1/relationships/areas", obj.Areas.Links["self"].Href);
    Assert.Equal("https://localhost:44317/api/types/1/areas", obj.Areas.Links["related"].Href);
}

public class Types
{
    public string Id { get; set; }

    public string Name { get; set; }

    public string Description { get; set; }

    public Relationship Areas { get; set; }
}
```

**Comment by jimmyrose** [comment]
Thanks Alex. I had used the above, but now if there _is_ data included for that relationship, I'd like it accessible as a List<Area>. 

Relationship<TData> only has the links and meta, the data doesn't get deserialized. 

If it looks like:

`public class Type
{
    public string Id { get; set; }

    public string Name { get; set; }

    public string Description { get; set; }

    public List<Area> { get; set; }
}`

and the relationship has no data, I would have expected the List<Area> to be empty, rather than throwing an exception that data is expected. 

If I use Regex to add an empty "data" element after the"links" element in the relationship in my example, it deserialises OK with an empty list. 

In [this method](https://github.com/codecutout/JsonApiSerializer/blob/87a54536e948aa434c63a5344df663a0178661c4/src/JsonApiSerializer/Util/ReaderUtil.cs#L188) it appears to be looking for "data" item, and it doesn't find one

**Comment by alex-davies** [comment]
Yes that should deserialise with a null, this looks to be a bug. Will try to get a fix out within next few days

In the mean time you should be able to use`Relationship<List<Area>> Areas`. This will give you access to the relationship links as well as to the `data` property if it is present

**Comment by jimmyrose** [comment]
Awesome cheers Alex. Thanks for the library it has been super helpful

**Comment by alex-davies** [comment]
This is now fixed in 1.6.1 (by https://github.com/codecutout/JsonApiSerializer/pull/86)
